### DISCUSSION TITLE
Docker daemon fails to start due to miss n folder in home directory

### DISCUSSION THREAD

**Discussion by adinunzio10** [canonical/docker-snap#246] [General]
I am not sure if this is unique to snap docker but today I ran into some issues after a power outage where my docker daemon refused to start. After an hour of troubleshooting I found that a folder for a docker plugin on my home drive vanished. I recreated the folder and bam docker is back to normal. 

How does 1 missing folder take out the whole daemon? Shouldn’t it be able to create the missing folder?

**Comment by locnnil** [comment]
Hi, thanks for your reporting.

Could you point what folder was missing? 

## Workaround when snap is corrupted

To avoid having corrupted snaps, you can [make a snapshot](https://snapcraft.io/blog/how-to-create-snapshots-of-your-snaps) of your docker installation: 

```
sudo snap save docker
```
This way, if your snap is corrupted somehow, you can restore it from the snapshot:

- First list the available snapshots:
```console
$ snap saved
Set  Snap    Age    Version  Rev   Size    Notes
9    docker  4m25s  24.0.5   2932  2.83MB  -
```
- Then use the number output under `Set` to restore it:

```console
$ sudo snap restore 9
Restored snapshot #9.
```

Or either completely uninstall it and install again:
```bash
sudo snap remove --purge docker && sudo snap install docker
```
